src/app/shared/directives/hide-if-not-authenticated.directive.ts

```diff
@@ -0,0 +1,90 @@
+import { ChangeDetectorRef, Directive, HostListener, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, TemplateRef, ViewContainerRef } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { AuthenticationService } from '../services/authentication.service';
+
+
+@Directive({
+  selector: '[appHideIfNotAuthenticated]'
+})
+export class HideIfNotAuthenticatedDirective implements OnInit, OnChanges {
+
+  constructor(
+    private templateRef: TemplateRef<unknown>,
+    private viewContainer: ViewContainerRef,
+    private authService: AuthenticationService
+  ) {}
+
+
+  @HostListener('click')
```

Comment:
I don`t think that you need to listen to click event in this directive

Reply:
yes, I removed it